pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -207,15 +309,31 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 		// - doing that would require significant refactoring to reflector
 		// we limit ourselves to just quickly stop the reflector here.
 		if c.isImmutable(object) {
+			item.markImmutable()
 			if item.stop() {
 				klog.V(4).Infof("Stopped watching for changes of %q/%q - object is immutable", namespace, name)
 			}
 		}
+
+		item.updateLastAccessTime(c.clock.Now())
 		return object, nil
 	}
 	return nil, fmt.Errorf("unexpected object type: %v", obj)
 }
 
+func (c *objectCache) startRecycleIdleWatch() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	for key, item := range c.items {
+		if !item.isStopped() && !item.isIdleTimeExceeded(c.clock.Now(), c.maxIdleTime) {
```

Comment:
You basically acquire mutex 3 times here (isStopped, isIdel, stop().

Instead, let's introduce a method `stopIfIdle(<current time>) that will do everything